menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -45,9 +48,47 @@ class SchemaController @Autowired()(
   extends VersionedModelController(schemaService) {
 
   import za.co.absa.enceladus.menas.utils.implicits._
-
   import scala.concurrent.ExecutionContext.Implicits.global
 
+  @PostMapping(Array("/remote"))
+  @ResponseStatus(HttpStatus.CREATED)
+  def handleRemoteFile(@AuthenticationPrincipal principal: UserDetails,
+                       @RequestParam remoteUrl: String,
+                       @RequestParam version: Int,
+                       @RequestParam name: String,
+                       @RequestParam format: Optional[String]): CompletableFuture[Option[Schema]] = {
+
+    val schemaType = SchemaType.fromOptSchemaName(format)
+    val (url, fileContent, mimeType) = {
+      try {
+        val url = new URL(remoteUrl)
+        val connection = url.openConnection()
+        val mimeType = connection.getContentType
+        val fileStream = Source.fromInputStream(connection.getInputStream)
+        val fileContent = fileStream.mkString
+        fileStream.close()
+
+        (url, fileContent, mimeType)
+
+      } catch {
+        case NonFatal(e) =>
+          throw RemoteSchemaRetrievalException(schemaType, s"$remoteUrl is currently unreachable to download a schema file from.", e)
```

Comment:
```suggestion
          throw RemoteSchemaRetrievalException(schemaType, s"Could not retrieve a schema file from $remoteUrl. Please check the correctness of the URL and a presence of the schema at the mentioned endpoint", e)
```

"Is unreachable" invokes in me "network issue and nothing else". What do you think about this proposition?

Reply:
Sure, it sounds better.